be/src/olap/rowset/beta_rowset_reader.cpp

```diff
@@ -174,7 +179,15 @@ Status BetaRowsetReader::get_segment_iterators(RowsetReaderContext* read_context
                                                            should_use_cache));
 
     // create iterator for each segment
-    for (auto& seg_ptr : _segment_cache_handle.get_segments()) {
+    auto& segments = _segment_cache_handle.get_segments();
+    auto [seg_start, seg_end] = segment_offset;
+    if (seg_start == seg_end) {
```

Comment:
Is there any possible that current rowset reader is allocated [seg_start=10,seg_end=10], then seg_start == seg_end, seg_start and seg_end is reset to [0, segments.size()]

Reply:
No. if spilit, unless rowset segment is empty. the segment_offset must have one segment by split